### PR TITLE
feat: add invalid refund page

### PR DESCRIPTION
**Pull request by DiegoHBustamante**
This PR contains the following changes:

- View for possible errors in the reimbursement process.
- Modifications in the refund methods to handle errors.
- Redirection to the invalidRefund view in case of error.

>Before the changes
Captura de pantalla 2024-11-18 a la(s) 6 18 27 p  m

> After the changes
Captura de pantalla 2024-11-21 a la(s) 4 22 07 p  m

### CODE REVIEW

File: src/app/invalid/invalidRefund.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+];
+export type InvalidViewProps = {
+  errorType: string; // Mensaje de error recibido
+}
+export const InvalidView = ({ errorType }: InvalidViewProps) => {
+  return (
+    <>
+      <Head>
+        <title>Transbank SDK Node - Error en Reembolso</title>
+      </Head>
```
[the remark this answers is not in the document]
Comment (DiegoHBustamante) [reply src/app/invalid/invalidRefund.tsx]: Changes were made

File: src/app/lib/webpay-mall-diferido/data.ts [hunk 2]
```diff
@@ -151,11 +151,22 @@ export const refundTransaction = async (
...
+    const refundResponse = await new WebpayPlus.MallTransaction(
+      getWebpatMallDeferredOptions()
+    ).refund(token_ws, buyOrder, commerceCode, amount);
 
-  return refundResponse;
+    return {
+      success: true,
+      refundResponse,
+    };
+  } catch (error: any) {
```
[the remark this answers is not in the document]
Comment (DiegoHBustamante) [reply src/app/lib/webpay-mall-diferido/data.ts]: any deleted

File: src/app/invalid/invalidRefund.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import Head from "next/head";
+import { Route } from "@/types/menu";
+import { Layout } from "@/components/layout/Layout";
+import { LayoutContent } from "@/components/layout/LayoutContent";
+import { Snippet } from "@/components/snippet/Snippet";
+
+
+// Breadcrumb para navegación
```
[the remark this answers is not in the document]
Comment (DiegoHBustamante) [reply src/app/invalid/invalidRefund.tsx]: deleted commentary

File: src/components/customError/CustomError.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import { Snippet } from "@/components/snippet/Snippet";
+
+export type InvalidViewProps = {
+  errorMessage: string;
+  actualBread: Route[]; 
+}
+export function CustomError({ errorMessage, actualBread }: InvalidViewProps){
+  return (
+      <Layout
+        pageTitle="Ocurrió un error"
```
Comment (mastudillot) [inline src/components/customError/CustomError.tsx:14]:
```suggestion
        pageTitle="Se ha producido un error"
```

File: src/components/customError/CustomError.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        pageDescription={
+          <span>    
+            Lo sentimos, no fue posible realizar la operación solicitada.
```
Comment (mastudillot) [inline src/components/customError/CustomError.tsx:17]:
```suggestion
            Se ha producido un error durante la operación y la API ha retornado un error.
```

**Review by mvarlic — APPROVED** [review]
LGTM!

File: src/components/customError/CustomError.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+          </span>
+        }
+        actualBread={actualBread}
+        activeRoute={actualBread[actualBread.length - 1]?.path}
+        additionalContent={
+          <div className="error-details">
+            <LayoutContent
+              pageTitle="¿Por qué no fue posible realizar la operación?"
+              pageDescription="Te proporcionamos una descripción detallada del error ocurrido, según la información proporcionada por el SDK."
```
Comment (mastudillot) [inline src/components/customError/CustomError.tsx:26]:
```suggestion
              pageDescription="El SDK captura el error retornado por la API, acontinuación se muestra el detalle de la excepción."
```

**Review by mastudillot — COMMENTED** [review]
Check my comments

**Review by mastudillot — APPROVED** [review]
LGTM

**Pull request merged**